core/internal/leet/livestore_test.go

```diff
@@ -0,0 +1,465 @@
+package leet_test
+
+import (
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/wandb/wandb/core/internal/leet"
+	"github.com/wandb/wandb/core/internal/transactionlog"
+	"github.com/wandb/wandb/core/pkg/leveldb"
+	spb "github.com/wandb/wandb/core/pkg/service_go_proto"
+	"google.golang.org/protobuf/proto"
+)
+
+// TestNewLiveStore_ValidFile tests creating a LiveStore with a valid file
+func TestNewLiveStore_ValidFile(t *testing.T) {
+	tmpFile, err := os.CreateTemp(t.TempDir(), "valid-*.wandb")
+	if err != nil {
+		t.Fatalf("CreateTemp: %v", err)
+	}
+	tmpFile.Close()
+	// transactionlog.Writer doesn't want the file to exist
+	_ = os.Remove(tmpFile.Name())
+
+	// Write a valid header to a tranaction log.
+	w, err := transactionlog.OpenWriter(tmpFile.Name())
+	if err != nil {
+		t.Fatalf("Failed to open transaction log for writing: %v", err)
+	}
+	w.Close()
+
+	// Now open with LiveStore
+	ls, err := leet.NewLiveStore(tmpFile.Name())
+	if err != nil {
+		t.Fatalf("NewLiveStore failed: %v", err)
+	}
+	defer ls.Close()
+
+	// Should be able to read (will get EOF since no records)
+	_, err = ls.Read()
+	if !errors.Is(err, io.EOF) {
+		t.Fatalf("Expected EOF, got: %v", err)
+	}
+}
+
+// TestNewLiveStore_NonExistentFile tests error handling for missing files
+func TestNewLiveStore_NonExistentFile(t *testing.T) {
+	_, err := leet.NewLiveStore("/nonexistent/path/file.wandb")
```

Comment:
```suggestion
	path := filepath.Join(t.TempDir(), "nonexistent.wandb")

	_, err := leet.NewLiveStore(path)
```

Is more precise. Otherwise it's not obvious whether this would test the right thing on Windows.

Maybe also assert on the error type? `assert.ErrorIs(t, err, os.ErrNotExist)`?